### ISSUE TITLE
Adding checkstyle to a libGDX project

### ISSUE THREAD

**Issue by Glusk2** [libgdx/libgdx#5581]
#### Issue details
Unable to add `checkstyle` to a libgdx project.

I've ran into this issue on my own project and decided to ask a question on stackoverflow: https://stackoverflow.com/questions/54851229/how-to-add-checkstyle-to-a-libgdx-project

Then I created a fresh project and the issue still persists.

#### Reproduction steps/code

1. Install Java 8
2. Install Android SDK
3. `cd <path-to-android-sdk>`
4. `sdkmanager "platforms;android-28" "platform-tools" "build-tools;28.0.3"`
5. Edit  the root `build.gradle` like so:
```
// ...
allprojects {
    // ..
    apply plugin: "checkstyle"
    // ...
// ...
```
6. Add `testProject/config/checkstyle/checkstyle.xml`
7. Run: `grawdlew clean build connectedCheck`

#### Version of LibGDX and/or relevant dependencies
```
gdxVersion = '1.9.9'
roboVMVersion = '2.3.5'
box2DLightsVersion = '1.4'
ashleyVersion = '1.7.0'
aiVersion = '1.8.0'
```

#### Stacktrace
```java
> Configure project : 
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Task :core:compileJava 
warning: [options] bootstrap class path not set in conjunction with -source 1.6
1 warning

> Task :android:compileJava FAILED
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:3: error: package android.os does not exist
import android.os.Bundle;
                 ^
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:9: error: cannot access Activity
public class AndroidLauncher extends AndroidApplication {
       ^
  class file for android.app.Activity not found
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:11: error: cannot find symbol
        protected void onCreate (Bundle savedInstanceState) {
                                 ^
  symbol:   class Bundle
  location: class AndroidLauncher
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:10: error: method does not override or implement a method from a supertype
        @Override
        ^
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:12: error: cannot find symbol
                super.onCreate(savedInstanceState);
                ^
  symbol:   variable super
  location: class AndroidLauncher
<path-to-project>testProject\android\src\com\github\glusk2\AndroidLauncher.java:14: error: cannot find symbol
                initialize(new Sprouts(), config);
                ^
  symbol:   method initialize(Sprouts,AndroidApplicationConfiguration)
  location: class AndroidLauncher
6 errors


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':android:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
61 actionable tasks: 60 executed, 1 up-to-date
```

#### Please select the affected platforms
- [x] Android
- [x] iOS (robovm)
- [ ] iOS (MOE)
- [x] HTML/GWT
- [x] Windows
- [x] Linux
- [x] MacOS

**Comment by NathanSweet** [comment]
This seems to be a configuration problem (gradle?) not a problem with libgdx.

**Comment by Glusk2** [comment]
It was a configuration problem: https://stackoverflow.com/a/55398135/6367213
I think  **LibGDX Project setup tool** should be updated and throw out Eclipse config for android projects.

**Comment by Glusk2** [comment]
This seems to have been resolved in #5700
